### PR TITLE
chore: show tooltip when hovering over abbreviation

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Create list of common abbreviations
- Change `mkdocs.yml` config file so the tooltip feature can work [^tooltips]

## Context:

It's easy to forget what abbreviations stand for. How about showing a tooltip when you hover over an abbreviation?

### What abbreviations should have a tooltip?

Maybe we don't _need_ to define really common abbreviations like JSON, HTTP and HTTPS: [^quote]

> ## If everyone knows an abbreviation, use it without explanation
> 
> There’s a short list of abbreviations that have entered common usage. When you use them, don’t define them—you’re just taking up space and annoying your user. Make sure the abbreviation you’re using is on the list. Examples include IBM, ATM, BMW, PhD, CIA, and FBI.
> 
> A closely related guideline is, “don’t define something that’s obvious to the user.” Most federal agencies, when writing a letter responding to an inquiry, insist on defining the agency name, as in, “Thank you for writing to the Federal Aviation Administration (FAA) about your concerns.” The letterhead says the name of the agency. The person wrote to the agency, and now the agency is writing back. The user is not going to be confused about what FAA means!

[^quote]: https://www.plainlanguage.gov/guidelines/words/minimize-abbreviations/
[^tooltips]: https://squidfunk.github.io/mkdocs-material/reference/tooltips/

### CODE REVIEW

File: includes/abbreviations.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!-- Material for MkDocs shows a tooltip when you hover over any abbreviation in this list. -->
+<!-- https://squidfunk.github.io/mkdocs-material/reference/tooltips/#adding-a-glossary -->
+
+<!-- Please keep this list sorted from A-Z. -->
+
+*[API]: Application Programming Interface
+*[AWS]: Amazon Web Services
+*[CD]: Continuous Development
```
Comment (rarkins) [inline includes/abbreviations.md:8]:
```suggestion
*[CD]: Continuous Deployment
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: includes/abbreviations.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<!-- Material for MkDocs shows a tooltip when you hover over any abbreviation in this list. -->
```
Comment (HonkingGoose) [inline includes/abbreviations.md:1]: I'm still thinking about this recommendation:

> If everyone knows an abbreviation, use it without explanation

I've basically added all abbreviations I could find into the list. But maybe we should drop really common abbreviations. But what's common depends on the previous experience of the reader... 😄 

What kind of assumptions are we making about the readers of our documentation? Do we aim for:

- Seasoned developers
- People new to programming, but who want to use a dependency bot
- Team managers who review/approve new bots before developers can use/install them
- Somebody else I'm missing to list

Also there's something to be said for being consistent, and just shoving all abbreviations in the list... 😄 

What do you think?

Comment (rarkins) [reply includes/abbreviations.md:1]: I think we can assume a reasonable maturity. For example they know API and JSON

File: includes/abbreviations.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+*[CD]: Continuous Deployment
+*[CI]: Continuous Integration
+*[CLI]: Command-Line Interface
+*[CVE]: Common Vulnerabilities and Exposures
+*[DCO]: Developer Certificate of Origin
+*[GCR]: Google Container Registry
+*[GPG]: GNU Privacy Guard
+*[HTML]: HyperText Markup Language
+*[HTTP]: Hypertext Transfer Protocol
+*[HTTPS]: Hypertext Transfer Protocol Secure
```
Comment (HonkingGoose) [inline includes/abbreviations.md:17]: Can we assume our readers know:

- HTML
- HTTP
- HTTPS (secure version of HTTP)

```suggestion
```

File: includes/abbreviations.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+*[JRE]: Java Runtime Environment
+*[LTS]: Long Term Support
+*[OCI]: Open Container Initiative
+*[PAT]: Personal Access Token
+*[SDK]: Software Development Kit
+*[SemVer]: Semantic Versioning
+*[SSH]: Secure Shell
+*[SSL]: Secure Sockets Layer
+*[TLS]: Transport Layer Security
+*[URL]: Uniform Resource Locator
```
Comment (HonkingGoose) [inline includes/abbreviations.md:29]:
```suggestion
```

We can probably drop URL as well?

**Review by HonkingGoose — COMMENTED** [review]
If you want to drop these abbreviations, please apply the suggestions. 😉

**Comment by HonkingGoose** [comment]
I'm messing something up with the Git submodule while merging the latest `main` into my branch... I marked this PR as "draft" so we don't merge it accidentally.

I'll keep this PR around until we have settled on the abbreviations list. After that, I'll just re-do my PR on a new branch. That way I don't break things when we merge. 😄

**Review by rarkins — APPROVED** [review]

**Pull request closed without merging**